gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/JdbcExtractor.java

```diff
@@ -914,15 +919,37 @@ public long getCount(CommandOutput<?, ?> response) throws RecordCountException {
     try {
       final ResultSetMetaData resultsetMetadata = resultset.getMetaData();
       final int columnCount = resultsetMetadata.getColumnCount();
-
+      Base64.Encoder encoder = Base64.getEncoder();
+      
       int batchSize = this.workUnit.getPropAsInt(ConfigurationKeys.SOURCE_QUERYBASED_FETCH_SIZE, 0);
       batchSize = (batchSize == 0 ? ConfigurationKeys.DEFAULT_SOURCE_FETCH_SIZE : batchSize);
 
       int recordCount = 0;
+      Set<String> blobDataNames = new HashSet<String>();
+      boolean initialed_blobDataNames = false;
       while (resultset.next()) {
+        if (!initialed_blobDataNames){
```

Comment:
Why do we have to initialize the set of blob columns inside the loop? Isn't it possible to do it outside the loop. In that case, the initialed_blobDataNames should not be needed.

nit: I assume you mean initializedBlobDataNames not initialed_blobDataNames .